libs/s25main/Loader.h

```diff
@@ -188,6 +187,8 @@ class Loader : public Singleton<Loader, SingletonPolicies::WithLongevity>
         RTTR_Assert(!item || res);
         return res;
     }
+    Log& logger_;
```

Comment:
I can't recommend storing a reference here, same problem, you don't know when it will be destroyed.
Use a shared_ptr on `ILog`

Reply:
References are non-owning references to existing objects. shared_ptr implies ownership and even worse: multiple ownerships. So references are appropriate here (besides being the least amount of change)

Reply:
but we have an implicit ownership here now. and you can't mock it.

of course its the least amount of change, but we should reconsider that on the next step then

Reply:
No ownership is here. That is the whole model of references: You don't own it, you reference it
And you can mock with references just as well.